### ISSUE TITLE
Incorrect literal type widening in function calls

### ISSUE THREAD

**Issue by tjf801** [microsoft/pyright#4895]
**Describe the bug**
If one generic literal type gets widened, all get widened, when that is not the expected behavior.

**Expected behavior**
`Matrix[Literal[0], Literal[3], Literal[4]]` should be able to be assigned to `Matrix[int, Literal[3], Literal[4]]` iff `Matrix` is covariant in its first parameter.

**Screenshots or Code**
```python
from typing import Generic, Literal, TypeVar

_T_co = TypeVar('_T', covariant=True)
_N = TypeVar('_N', bound=int)
_M = TypeVar('_M', bound=int)

class Matrix(Generic[_T_co, _N, _M]): ...

def zeros(__n: _N, __m: _M, /) -> Matrix[Literal[0], _N, _M]: ...


# no error
A: Matrix[Literal[0], Literal[3], Literal[4]] = zeros(3, 4)

# unexpected: "int" cannot be assigned to type "Literal[3]"
B: Matrix[int, Literal[3], Literal[4]] = zeros(3, 4)

# expected: "Matrix[Literal[0], Literal[3], Lieral[4]]" is incompatible with "Matrix[int, int, int]"; TypeVar "_N@Matrix" is invariant; "Literal[3]" is incompatible with "int"
C: Matrix[int, int, int] = zeros(3, 4)
```

**VS Code extension or command-line**
I am using the Pylance VS Code extension.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.302, which I just published. It will also be included in a future release of pylance.
